includes/acl/compression/impl/compact_constant_streams.h

```diff
@@ -149,10 +342,195 @@ namespace acl
 					bone_range.scale = track_stream_range::from_min_extent(scale, rtm::vector_zero());
 
 					num_default_bone_scales += bone_stream.is_scale_default ? 1 : 0;
+
+#ifdef ACL_COMPRESSION_OPTIMIZED
+
+					has_constant_bone_scales = true;
+					ACL_ASSERT(raw_segment.bone_streams[bone_index].scales.get_num_samples() == num_samples, "Raw scale sample mismatch!");
+
+#endif
+
 				}
 			}
 
 			context.has_scale = num_default_bone_scales != num_bones;
+
+#ifdef ACL_COMPRESSION_OPTIMIZED
+
+			deallocate_type_array(allocator, max_adjusted_shell_distances, num_bones);
```

Comment:
Could benefit from a few comments to explain what is going on.
If I'm not mistaken, we apply error correction to our constant sub-tracks by transforming everything into object space and back into local space.

Reply:
hat's the gist of it. We use object space of the original data as ground truth, and only deviate for 2 reasons, and as briefly as possible.

Replace an original local value with a new constant value.
Correct for the manipulation of an original local value by an ancestor ASAP.